public/pages/CreateIndex/components/IndexMapping/IndexMapping.tsx

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { forwardRef, useCallback, useState, Ref, useRef, useMemo, useImperativeHandle } from "react";
+import { EuiTreeView, EuiIcon, EuiTreeViewProps, EuiButton, EuiSpacer, EuiButtonGroup, EuiLink } from "@elastic/eui";
+import { set, get, isEmpty } from "lodash";
+import JSONEditor, { IJSONEditorRef } from "../../../../components/JSONEditor";
+import { Modal } from "../../../../components/Modal";
+import { MappingsProperties, MappingsPropertiesObject } from "../../../../../models/interfaces";
+import CustomFormRow from "../../../../components/CustomFormRow";
+import MappingLabel, { IMappingLabelRef } from "../MappingLabel";
+import "./IndexMapping.scss";
+
+export const transformObjectToArray = (obj: MappingsPropertiesObject): MappingsProperties => {
+  return Object.entries(obj).map(([fieldName, fieldSettings]) => {
+    const { properties, ...others } = fieldSettings;
+    const payload: MappingsProperties[number] = {
+      ...others,
+      fieldName,
+    };
+    if (properties) {
+      payload.properties = transformObjectToArray(properties);
+    }
+    return payload;
+  });
+};
+
+export const transformArrayToObject = (array: MappingsProperties): MappingsPropertiesObject => {
```

Comment:
Kind of confusing there is both a transformArrayToObject and transformArrayToObj. Maybe add some comments?

Reply:
Good point, fixed on https://github.com/opensearch-project/index-management-dashboards-plugin/commit/3852583062f4d60385539ddd977ea1f310ab9291